File: Smallest Positive missing number - GFG/smallest-positive-missing-number.py
#User function Template for python3

class Solution:
    
    #Function to find the smallest positive number missing from the array.
    def missingNumber(self,arr,n):
        #Your code here
        dic = {}
        for ele in arr:
            if ele in dic:
                dic[ele] += 1
            else:
                dic[ele] = 1
        for i in range(1, len(arr)+1):
            if i not in dic:
                return i
        return n+1

#{ 
 # Driver Code Starts
#Initial Template for Python 3


import math


def main():
        T=int(input())
        while(T>0):
            
            n=int(input())
            
            arr=[int(x) for x in input().strip().split()]
            
            ob=Solution()
            print(ob.missingNumber(arr,n))
            
            T-=1


if __name__ == "__main__":
    main()
# } Driver Code Ends

File: Smallest Positive missing number - GFG/smallest-positive-missing-number.java
//{ Driver Code Starts
import java.util.*;


// } Driver Code Ends

class Solution
{
    //Function to find the smallest positive number missing from the array.
    static int missingNumber(int arr[], int size)
    {
        // Your code here
        HashMap<Integer, Integer> mp = new HashMap<>();
        for(int ele: arr){
            if(mp.containsKey(ele)){
                mp.put(ele, mp.get(ele)+1);
            }
            else mp.put(ele, 1);
        }
        for(int i = 1 ; i <= size ; i++){
            if(mp.containsKey(i) == false){
                return i;
            }
        }
        return size+1;
    }
}


//{ Driver Code Starts.

class Main
{   
    public static void main (String[] args) 
    {

		Scanner sc=new Scanner(System.in);
		
        //taking testcases
		int t=sc.nextInt();
		while(t-->0){
		    
		    //input number n
			int n=sc.nextInt();
			int[] arr=new int[n];
			
			//adding elements to the array
			for(int i=0;i<n;i++)
				arr[i]=sc.nextInt();
				
			Solution obj = new Solution();
			
			//calling missingNumber()
			int missing = obj.missingNumber(arr,n);
			System.out.println(missing);
		}
    }
}


// } Driver Code Ends